extension/src/background/helpers/dataStorage.ts

```diff
@@ -80,6 +81,13 @@ export const normalizeMigratedData = async () => {
   const localStore = dataStorageAccess(browserLocalStorage);
   const localStorageEntries = Object.entries(localStorage);
 
+  const applicationState = await localStore.getItem(APPLICATION_ID);
+  const isLocalStoreSetup = !!applicationState?.length;
```

Comment:
checking the existence of the `applicationState` to determine if this migration has been run. `applicationState` is a base setting that is set right away in user flow, so we should be able to rely on the existence of this to determine that the user has interacted with Freighter before

Reply:
The interface we set up for chrome.storage.local only allows getting by string key, it doesn't allow you to get the whole object by passing `null`. Because of that, I decided to check by key, which is a more specific check anyway and should be more deterministic